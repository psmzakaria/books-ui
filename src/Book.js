import React, { Component } from "react";
const API_HOST = process.env.REACT_APP_BOOKS_API || "http://localhost:3000";
class Book extends Component {
  constructor() {
    super();
    this.state = {
      books: []
    };
  }
  componentDidMount() {
    this.getBooks("");
  }

  getBooks = async search => {
    const url = `${API_HOST}/books`;
    // const response = await fetch(url);
    // const bookData = await response.json();
    // this.setState({ books: bookData });

    this.setState({
      books: [
        {
          title: "HARRY POTTER",
          _id: "23094823048"
        }
      ]
    });
  };

  render() {
    return (
      <div>
        {this.state.books.map((book) => {
          return <li key ={book._id}> {book.title}</li>;
        })}
      </div>
    );
  }
}

export default Book;
