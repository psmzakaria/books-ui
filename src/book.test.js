import React from "react";
import ReactDOM from "react-dom";
import Book from "./Book";

import ShallowRenderer from "react-test-renderer/shallow";

it("renders title and boollist after the component mount", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Book />);
  const instance = renderer.getMountedInstance();
  instance.componentDidMount();

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});

it("renders title and authorlist after the component mount", async () => {
  fetch.mockResponseOnce(
    JSON.stringify([
      {
        Title: "Charlie and chcolate factory",
        _id: "478957854897"
      },
      {
        name: "Dolly Lama a Good life ",
        _id: "473895869547"
      }
    ])
  );

  const renderer = new ShallowRenderer();
  renderer.render(<Book />);
  const instance = renderer.getMountedInstance();
  await instance.componentDidMount();
  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});
