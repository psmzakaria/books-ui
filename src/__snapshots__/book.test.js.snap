// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders title and authorlist after the component mount 1`] = `
<div>
  <li>
     
    HARRY POTTER
  </li>
</div>
`;

exports[`renders title and boollist after the component mount 1`] = `
<div>
  <li>
     
    HARRY POTTER
  </li>
</div>
`;
